odbc/src/api/driver_connect_tests.rs

```diff
@@ -0,0 +1,150 @@
+use crate::{handles::definitions::*, SQLDriverConnectW, SQLGetDiagRecW};
+use constants::{NOT_IMPLEMENTED, NO_DSN_OR_DRIVER, UNABLE_TO_CONNECT};
+use odbc_sys::{DriverConnectOption, HandleType, SqlReturn};
+use std::sync::RwLock;
+
+mod unit {
+    use super::*;
+    use odbc_sys::SmallInt;
+    #[test]
+    fn connect_edge_cases() {
```

Comment:
It would probably be better to extract each test so that the tests don't stop on the first failed assertion.

Reply:
That's a good idea.  I changed the function `driver_connect_check_diagnostics()` into a macro and now each of the tests here are run separately.

Reply:
Perfect. Thank you :)